src/local_scheduler/local_scheduler.cc

```diff
@@ -147,6 +147,9 @@ void LocalSchedulerState_free(LocalSchedulerState *state) {
   /* Reset the SIGTERM handler to default behavior, so we try to clean up the
    * local scheduler at most once. If a SIGTERM is caught afterwards, there is
    * the possibility of orphan worker processes. */
+  /* TODO(swang): Add a null heartbeat that tells the global scheduler that we
+   * are dead. This avoids having to wait for the timeout before marking us as
+   * dead in the db_client table, in cases where we can do a clean exit. */
```

Comment:
For the time being, this will add 10 seconds or so to each of our fault tolerance tests, right?

Reply:
Yes.